### PR TITLE
Update `OracleContainer` and re-activate on CI

### PR DESCRIPTION
**Pull request by kiview**
Trying to reproduce #4297

### CODE REVIEW

**Comment by kiview** [comment]
The changes from the test class should be moved to the real Oracle module container class and then we can enable the CI in the tests.

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 1]
```diff
@@ -1,45 +1,40 @@
...
-public class OracleContainer extends JdbcDatabaseContainer<OracleContainer> {
+public class OracleContainer<SELF extends OracleContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
 
     public static final String NAME = "oracle";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("gvenzl/oracle-xe");
+
+    @Deprecated
+    public static final String DEFAULT_TAG = "18.4.0";
+    @Deprecated
+    public static final String IMAGE = DEFAULT_IMAGE_NAME.getUnversionedPart();
```
Comment (kiview) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java:20]: Why add those public constants, if they were not present in the first place?

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 2]
```diff
@@ -88,19 +83,28 @@ public String getPassword() {
     }
 
     @Override
-    public OracleContainer withUsername(String username) {
+    public SELF withUsername(String username) {
+        if (StringUtils.isEmpty(username)) {
+            throw new IllegalArgumentException("Username cannot be null or empty");
+        }
+        if (ORACLE_SYSTEM_USERS.contains(username.toLowerCase())) {
```
Comment (kiview) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java:90]: This is nice 👍

**Review by kiview — COMMENTED** [review]
Can't approve because I opened the PR 😅 
LGTM, just note the comment.

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 3]
```diff
@@ -1,45 +1,40 @@
...
+import java.util.Arrays;
+import java.util.List;
 import java.util.Set;
 import java.util.concurrent.Future;
 
-/**
- * @author gusohal
- */
-public class OracleContainer extends JdbcDatabaseContainer<OracleContainer> {
+public class OracleContainer<SELF extends OracleContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
```
Comment (bsideup) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java:12]: no need for SELF-typing here:
```suggestion
public class OracleContainer extends JdbcDatabaseContainer< OracleContainer > {
```

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 4]
```diff
@@ -1,45 +1,40 @@
...
 
     public static final String NAME = "oracle";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("gvenzl/oracle-xe");
+
+    @Deprecated
+    public static final String DEFAULT_TAG = "18.4.0-slim";
+    @Deprecated
+    public static final String IMAGE = DEFAULT_IMAGE_NAME.getUnversionedPart();
```
Comment (bsideup) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java:20]: These are newly added fields - no need to make them public and deprecated

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 5]
```diff
@@ -122,4 +126,11 @@ public Integer getWebPort() {
     public String getTestQueryString() {
         return "SELECT 1 FROM DUAL";
     }
+
+    @Override
+    protected void configure() {
+        addEnv("ORACLE_PASSWORD", password);
+        addEnv("APP_USER", username);
+        addEnv("APP_USER_PASSWORD", password);
```
Comment (bsideup) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java:134]:
```suggestion
        withEnv("ORACLE_PASSWORD", password);
        withEnv("APP_USER", username);
        withEnv("APP_USER_PASSWORD", password);
```

**Pull request merged**